### ISSUE TITLE
[java] RuleSetFactoryCompatibility.java uses regular expressions which cause false positive warnings against rules that are commented out in the xml config file

### ISSUE THREAD

**Issue by darian-sanity** [pmd/pmd#393]
When using an XML config file while running PMD from the command line against our Java code, the RuleSetFactoryCompatibility.java reports warnings while running PMD for rules that have been renamed or moved etc. Using an XML comment does not prevent the warnings from being reported and causes user confusion.

An example can be reproduced by defining an XML config file with old rule names that are commented out with XML comment markers.

```
<!--
<rule ref="rulesets/java/basic.xml/EmptySynchronizedBlock"><priority>2</priority></rule>
-->
<rule ref="rulesets/java/empty.xml/EmptySynchronizedBlock"><priority>2</priority></rule>

WARNING: Applying rule set filter: The rule "EmptySynchronizedBlock" has been moved from ruleset "basic" to "empty". Please change your ruleset!
```

The problem appears to be due to regular expressions being used (to parse the XML config file) which do not know about XML commenting.

**Running PMD through:** *[CLI]*

**Comment by adangel** [comment]
Duplicate of #4378

Note: RuleSetFactoryCompatibility will be removed with PMD 7, see #4314 / #4795
